pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -360,18 +361,26 @@ func (lb *ListenerBuilder) buildVirtualOutboundListener(configgen *ConfigGenerat
 		isTransparentProxy = proto.BoolTrue
 	}
 
-	filterChains := buildOutboundCatchAllNetworkFilterChains(configgen, lb.node, lb.push)
+	_, localhost := getActualWildcardAndLocalHost(lb.node)
 
-	actualWildcard, _ := getActualWildcardAndLocalHost(lb.node)
+	filterStack := buildOutboundCatchAllNetworkFiltersOnly(lb.push, lb.node)
+	defaultFilterChain := &listener.FilterChain{
```

Comment:
Move the catchall filter chain out of `FilterChains` and into `DefaultFilterChain`. This is done because of Envoy's matching logic. If a filter chain specifies a destination port but the rest of the `FilterChainMatch` is not a match, Envoy _does not_ fallback to filter chains with wildcard ports. Instead, it falls back to the default filter chain. We need this now since the blackhole filter chain now has additional `FilterChainMatch` criteria beyond just destination port.

See https://github.com/envoyproxy/envoy/blob/main/source/server/filter_chain_manager_impl.cc#L482